collated/main/Kowalski985.md

```diff
@@ -0,0 +1,516 @@
+
+    private static final String[] commandsWithPrefixes = {"add", "edit"};
+
+    public static AutocompleteCommand getInstance(String commandName) {
+
+        switch (commandName) {
+        case AddCommand.COMMAND_WORD:
+            return ADD;
+
+        case ClearCommand.COMMAND_WORD:
+            return CLEAR;
+
+        case DeleteCommand.COMMAND_WORD:
+            return DELETE;
+
+        case DeleteTagCommand.COMMAND_WORD:
+            return DELETE_TAG;
+
+        case EditCommand.COMMAND_WORD:
+            return EDIT;
+
+        case ExitCommand.COMMAND_WORD:
+            return EXIT;
+
+        case FindCommand.COMMAND_WORD:
+            return FIND;
+
+        case HelpCommand.COMMAND_WORD:
+            return HELP;
+
+        case HistoryCommand.COMMAND_WORD:
+            return HISTORY;
+
+        case ImportCommand.COMMAND_WORD:
+            return IMPORT;
+
+        case ListCommand.COMMAND_WORD:
+            return LIST;
+
+        case RedoCommand.COMMAND_WORD:
+            return REDO;
+
+        case SelectCommand.COMMAND_WORD:
+            return SELECT;
+
+        case TabCommand.COMMAND_WORD:
+            return TAB;
+
+        case UndoCommand.COMMAND_WORD:
+            return UNDO;
+
+        default:
+            return NONE;
+        }
+    }
+
+    public static boolean hasIndexParameter (String command) {
+        return Arrays.asList(commandsWithIndexes).contains(command);
+    }
+
+    public static boolean hasPrefixParameter (String command) {
+        return Arrays.asList(commandsWithPrefixes).contains(command);
+    }
+
+
+}
+```
+###### \java\seedu\address\ui\Autocompleter.java
+``` java
+/**
+ * Autocomplete utility used in the command box
+ */
+public class Autocompleter {
+
+    private static final String PROMPT_USER_TO_USE_HELP_MESSAGE = "To see what commands are available, type 'help' "
+            + "into the command box";
+    private static final String MULTIPLE_RESULT_MESSAGE = "Multiple matches found";
+    private static final String EMPTY_STRING = "";
+    private static final String SPACE = " ";
+
+    private int resultIndex;
+    private int countingIndex;
+    private String textInCommandBox;
+    private AutocompleteState state;
+    private AutocompleteCommand currentCommand;
+    private ArrayList<String> possibleAutocompleteResults;
+    private CommandBoxParser parser;
+    private Logic logic;
+
+
+    public Autocompleter(Logic logic) {
+        registerAsAnEventHandler(this);
+        this.logic = logic;
+        parser = new CommandBoxParser();
+        resultIndex = 0;
+        countingIndex = 1;
+        textInCommandBox = EMPTY_STRING;
+        state = AutocompleteState.EMPTY;
+        currentCommand = AutocompleteCommand.NONE;
+        possibleAutocompleteResults = new ArrayList<>();
+    }
+
+    /**
+     * Returns the autocompleted command to be filled into the command box
+     * @return autocomplete text
```

Comment:
In this case there is no need for the @return tag since the javadoc description handles it.